src/contracts/core/AllocationManager.sol

```diff
@@ -53,13 +53,19 @@ contract AllocationManager is
     function slashOperator(
         SlashingParams calldata params
     ) external onlyWhenNotPaused(PAUSED_OPERATOR_SLASHING) {
+        // Assert that the amount to slash is within bounds of 1 and WAD (1e18).
         require(0 < params.wadToSlash && params.wadToSlash <= WAD, InvalidWadToSlash());
 
-        // Check that the operator is registered and slashable
         OperatorSet memory operatorSet = OperatorSet({avs: msg.sender, operatorSetId: params.operatorSetId});
-        bytes32 operatorSetKey = _encodeOperatorSet(operatorSet);
+
+        // Assert that the provided strategies are all whitelisted for the given operator set.
+        require(avsDirectory.isOperatorSetStrategyBatch(operatorSet, params.strategies), InvalidStrategy());
+        // Asser that the operator is slashable for the given operator set.
         require(avsDirectory.isOperatorSlashable(params.operator, operatorSet), InvalidOperator());
 
+        // Check that the operator is registered and slashable
+        bytes32 operatorSetKey = _encodeOperatorSet(operatorSet);
```

Comment:
Nit: comment isn't doing this check? I think you mean to put above `require(avsDirectory.isOperatorSlashable(params.operator, operatorSet), InvalidOperator());`